pkg/nfd-master/nfd-master.go

```diff
@@ -1425,6 +1469,10 @@ func (m *nfdMaster) filterFeatureAnnotations(annotations map[string]string) map[
 
 		outAnnotations[annotation] = value
 	}
+	if m.config.Restrictions.DisableAnnotations {
+		outAnnotations = map[string]string{}
+	}
```

Comment:
I think this is fine. But let's be consistent, either add logging (similarly to labels and ERs) like:

```suggestion
	if len(outAnnotations) > 0 && m.config.Restrictions.DisableAnnotations {
		klog.V(2).InfoS("node annotations are disabled in configuration (restrictions.disableAnnotations=true)")
		outAnnotations = map[string]string{}
	}
```

Or then remove logging from labels and ERs. WDYT?